### PR TITLE
1445387: Set locale fact to Unknown if value cannot be determined

### PR DESCRIPTION
**Pull request by kahowell**
That is if the language code from `getdefaultlocale`[1] is `None`, then
special case this as 'Unknown'.

[1]: https://docs.python.org/2/library/locale.html#locale.getdefaultlocale

### CODE REVIEW

File: src/rhsmlib/facts/host_collector.py [hunk 1]
```diff
@@ -67,7 +67,10 @@ def get_all(self):
         host_facts.update(firmware_info_dict)
 
         locale_info = {}
-        locale_info['system.default_locale'] = ".".join(locale.getdefaultlocale())
+        effective_locale = 'Unknown'
+        if locale.getdefaultlocale()[0] is not None:
```
Comment (cnsnyder) [inline src/rhsmlib/facts/host_collector.py]: Is it not possible for locale.getdefaultlocale() to return ("Something_not_none", None)?

Comment (kahowell) [inline src/rhsmlib/facts/host_collector.py]: Hrmmm. Good question actually... I tried

```shell
$ LC_ALL=en_us python -c 'import locale; print locale.getdefaultlocale()'
('en_US', 'ISO8859-1')
```

but lemme code for that just in case...

File: test/rhsmlib_test/test_host_collector.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import mock
+
+from rhsmlib.facts import host_collector
+
+
+class HostCollectorTest(unittest.TestCase):
+
+    @mock.patch('locale.getdefaultlocale')
+    def test_unknown_locale(self, mock_locale):
```
Comment (cnsnyder) [inline test/rhsmlib_test/test_host_collector.py]: If we think it's useful, I would suggest adding one more test for the case of ("Not_None", None) being returned from locale.getdefaultlocale()

**Comment by kahowell** [comment]
387aef3..9c43204: filter out 'None' encoding (just in case)
